alsoHeresTheInterviewQuestionIHad/alsoHeresTheInterviewQuestionIHad.js

```diff
@@ -0,0 +1,31 @@
+const fs = require("fs");
+const streetArray = fs.readFileSync("data.txt", "utf-8").split(",");
+let answer = "";
+let tallestBuilding = Math.max(...streetArray);
+let numberOfBuildings = streetArray.length;
+let tallestBuildingSoFar = 0;
+let bestArea = tallestBuilding * numberOfBuildings;
+let bannerOne = [tallestBuilding, numberOfBuildings];
+let bannerTwo = [tallestBuilding, numberOfBuildings];
+let bestBannerOne = [tallestBuilding, numberOfBuildings];
+let bestBannerTwo = [tallestBuilding, numberOfBuildings];
+for (let i = 0; i < numberOfBuildings - 1; i++){
+    if (streetArray[i] > tallestBuildingSoFar) {
+        tallestBuildingSoFar = streetArray[i];
+    }
+    bannerOne = [tallestBuildingSoFar, i + 1];
+    bannerTwo = [Math.max(...streetArray.slice(i + 1, streetArray.length)), numberOfBuildings - i - 1];
+
+//    console.log("At address " + i + " the two banners required would be " + bannerOne + " and " + bannerTwo);
+
+    if (((bannerOne[0] * bannerOne[1]) + (bannerTwo[0] * bannerTwo[1])) < bestArea) {
+        bestArea = (bannerOne[0] * bannerOne[1]) + (bannerTwo[0] * bannerTwo[1]);
+        bestBannerOne = [bannerOne[0], bannerOne[1]];
+        bestBannerTwo = [bannerTwo[0], bannerTwo[1]];
+    }
+}
+answer = "The best area we can cover is " + bestArea + " and that is from a banner of " + bestBannerOne[0] + "x" + bestBannerOne[1];

```

Comment:
It's cool that you figured out how to log out the dimensions of the banner(s), but definitely not necessary. 
We just need to return the minimum sum of the banners that we were able to find. Ideally this means it's also in a function.

Reply:
Oh, I suppose I agree, but I did want to "show my work" and I also thought it would be useful in debugging to know what banners my code was thinking.  I will deprecate those arrays.  Let me reflect on how I can meaningfully functionise something here.